internal/operator/cluster/pgbadger.go

```diff
@@ -0,0 +1,200 @@
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/crunchydata/postgres-operator/internal/kubeapi"
+	"github.com/crunchydata/postgres-operator/internal/operator"
+	"github.com/crunchydata/postgres-operator/internal/util"
+	crv1 "github.com/crunchydata/postgres-operator/pkg/apis/crunchydata.com/v1"
+
+	log "github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	// pgBadgerServicePortName is the name used to identify the pgBadger port in
+	// the service
+	pgBadgerServicePortName = "pgbadger"
+)
+
+// AddPGBadger ensures that a PostgreSQL cluster is able to undertake the
+// actions required by the "crunchy-badger", i.e. updating the Service.
+// This executes regardless if this is a standby cluster.
+//
+// This does not modify the Deployment that has the pgBadger sidecar. That is
+// handled by the "UpdatePGBadgerSidecar" function, so it can be handled as part
+// of a rolling update.
+func AddPGBadger(clientset kubernetes.Interface, restconfig *rest.Config, cluster *crv1.Pgcluster) error {
+	ctx := context.TODO()
+	// set up the Services, which are still needed on a standby
+	services, err := getClusterInstanceServices(clientset, cluster)
+	if err != nil {
+		return err
+	}
+
+	// loop over each service to perform the necessary modifications
+svcLoop:
+	for i := range services.Items {
+		svc := &services.Items[i]
+
+		// loop over the service ports to see if pgBadger port is already set up. if
+		// it is, we can continue and skip the outerloop
+		for _, svcPort := range svc.Spec.Ports {
+			if svcPort.Name == pgBadgerServicePortName {
+				continue svcLoop
+			}
+		}
+
+		// otherwise, we need to append a service port to the list
+		port, err := strconv.ParseInt(
+			util.GetValueOrDefault(cluster.Spec.PGBadgerPort, operator.Pgo.Cluster.PGBadgerPort), 10, 32)
+		// if we can't parse this for whatever reason, issue a warning and continue on
+		if err != nil {
+			log.Warn(err)
+		}
+
+		svcPort := v1.ServicePort{
+			Name:     pgBadgerServicePortName,
+			Protocol: v1.ProtocolTCP,
+			Port:     int32(port),
+		}
+
+		svc.Spec.Ports = append(svc.Spec.Ports, svcPort)
+
+		// if we fail to update the service, warn, but continue on
+		if _, err := clientset.CoreV1().Services(svc.Namespace).Update(ctx, svc, metav1.UpdateOptions{}); err != nil {
+			log.Warn(err)
+		}
+	}
+
+	return nil
+}
+
+// RemoveExporter disables the ability for a PostgreSQL cluster to run a
```

Comment:
RemovePGBadger

Reply:
Gee I wonder where I copy and pasted that from 🤔